src/pages/_app.tsx

```diff
@@ -44,6 +44,20 @@ export default function App({ Component, pageProps }: AppProps) {
     }
   }, [tab]);
 
+  const [rpcUrl, setRPCUrl] = useState<string>('https://neat-hidden-sanctuary.solana-mainnet.discover.quiknode.pro/2af5315d336f9ae920028bbb90a73b724dc1bbed');
+
+  const validateURL = (url: string) => {
+    const URLRegex =
+      /https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,}\b([-a-zA-Z0-9()@:%_\+.~#?&//=]*)/gi;
```

Comment:
this is fine but another way could be just do a try and catch with `new URL`

<img width="668" alt="image" src="https://user-images.githubusercontent.com/32449414/217196638-fdaf73d5-664b-4f84-9af6-944dcbb3deeb.png">

Reply:
sounds neater, but does not validate as much as I would've liked

new URL('https://google.')
new URL('https://a')
these url passes `new URL()`

Reply:
it throw error right? no?